### ISSUE TITLE
OpenCV 2.4.13.4 (unnecessarily) requires CMake 2.8.12.2 as minimum

### ISSUE THREAD

**Issue by gdsotirov** [opencv/opencv#9870]
##### System information (version)
- OpenCV: 2.4.13.4
- Operating System: Slackware Linux 14.0
- Compiler: GCC 4.7.1
- CMake: 2.8.8

##### Detailed description

The minimum required version of CMake changed from 2.6.3 to 2.8.12.2 from OpenCV 2.4.13.3 to 2.4.13.4 (see commit 3ba7c1667071679e8370719c80d30bf6de06cb23). This prevents building the newest patch version of the 2.4 branch for older OSes (such as Slackware 14.0, which only comes with CMake 2.8.8). However, this do not seem to be a real requirement, because OpenCV builds successfully with CMake 2.8.8 on Slackware 14.0, I confirm, although with some warnings in regards to Qt5 includes, but nothing new since 2.4.13.3.

##### Steps to reproduce

1. Download OpenCV 2.4.13.4 (`wget 'https://github.com/opencv/opencv/archive/2.4.13.4.tar.gz' -O opencv-2.4.13.4.tar.gz`);
2. Unpack the source archive (`tar -xvf opencv-2.4.13.4.tar.gz`);
3. Change to source directory (`cd opencv-2.4.13.4`);
4. Configure like this:
```mkdir -p build; cd build
cmake -LA \
      -DCMAKE_BUILD_TYPE=Release \
      -DCMAKE_INSTALL_PREFIX=/usr \
      -DCMAKE_LIBRARY_PATH=/usr/lib${LIBDIRSUFFIX} \
      -DOPENCV_DOC_INSTALL_PATH=/usr/doc/$PNAME-$VERSION \
      -DBUILD_SHARED_LIBS=ON \
      -DBUILD_PACKAGE=OFF \
      -DBUILD_NEW_PYTHON_SUPPORT=ON \
      -DBUILD_OCTAVE_SUPPORT=ON \
      -DBUILD_SWIG_PYTHON_SUPPORT=ON \
      -DBUILD_DOCS=ON \
      -DBUILD_EXAMPLES=ON \
      -DBUILD_TESTS=ON \
      -DENABLE_SSE=ON \
      -DENABLE_SSE2=ON\
      -DENABLE_SSE3=ON \
      -DENABLE_SSSE3=ON \
      -DENABLE_SSE41=ON \
      -DENABLE_SSE42=ON \
      -DENABLE_FAST_MATH=OFF \
      -DENABLE_OMIT_FRAME_POINTER=OFF \
      -DENABLE_PRECOMPILED_HEADERS=ON \
      -DOPENCV_BUILD_3RDPARTY_LIBS=OFF \
      -DINSTALL_C_EXAMPLES=ON \
      -DINSTALL_OCTAVE_EXAMPLES=ON \
      -DINSTALL_PYTHON_EXAMPLES=ON \
      -DCMAKE_C_FLAGS="$SLKCFLAGS" \
      -DCMAKE_CXX_FLAGS="$SLKCFLAGS -D__STDC_CONSTANT_MACROS" \
      -DWITH_1394=ON \
      -DWITH_EIGEN=ON \
      -DWITH_FFMPEG=ON \
      -DWITH_GSTREAMER=ON \
      -DWITH_GTK=ON \
      -DWITH_JPEG=ON \
      -DWITH_PNG=ON \
      -DWITH_QT=ON \
      -DWITH_TIFF=ON \
      -DWITH_V4L=ON \
      -DWITH_XINE=ON \
      ..
```

As a result the following error appears:

`CMake Error at CMakeLists.txt:44 (cmake_minimum_required):
  CMake 2.8.12.2 or higher is required.  You are running version 2.8.8`

It's easy to fix the CMakeLists.txt file, but I wonder shouldn't this be fixed upstream if there are going to be new releases in the 2.4 branch.

**Comment by alalek** [comment]
Supporting of modern/popular OS/compilers/libraries has higher priority than supporting of legacy configurations.
At least we are not able to test/develop for all of them (no free resources for that).

> Slackware 14.0
Slackware 14.2 as CMake 3.5+: https://mirrors.slackware.com/slackware/slackware-14.2/source/d/cmake/

BTW, Usually there is no huge problems to unpack/install custom CMake binaries for build-only activity.

**Comment by gdsotirov** [comment]
@alalek Yes, sure, but for Slackware 14.2 I'm already building OpenCV 3.3 and my aim is only to upgrade the version for Slackware 14.0 to the latest patch version (which presumably also shouldn't break dependencies). I understand that you do not have the resources to test/develop for all the various configurations, but as I wrote in my initial description of the issue the minimum requirement on CMake is not a real one (i.e. I was able to build successfully with 2.8.8). Anyway, I have an easy workaround for now with just patching the CMakeLists.txt file, so you could close the issue.

**Comment by gdsotirov** [comment]
https://github.com/opencv/opencv/blob/2.4.13.6/CMakeLists.txt#L39

Still not in OpenCV 2.4.13.6 :-(

**Comment by gdsotirov** [comment]
And still not fixed in OpenCV 2.4.13.7 :-(

https://github.com/opencv/opencv/blob/2.4.13.7/CMakeLists.txt#L39

**Comment by gdsotirov** [comment]
:-)
